### PR TITLE
acceptance test for shares received from multiple users with the same name file/folder

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

Includes acceptance tests for:

- shares received from multiple users with the same name file/folder

- shares received from multiple users with the same name file/folder and different permissions

- new user is created and added to a group. The group has shares that match the skeleton

## Related Issue

- Fixes <issue_link>
part of https://github.com/owncloud/core/issues/35414
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 1]
```diff
@@ -648,4 +648,72 @@ Feature: sharing
...
\ No newline at end of file
+
+  Scenario Outline: multiple users share a file with the same name but different permissions to a user
+    Given using OCS API version "<ocs_api_version>"
+    And these users have been created with default attributes and without skeleton files:
+      | username |
+      | user1    |
+      | user2    |
+      | user3    |
+    And user "user1" has uploaded file with content "user2 file" to "/randomfile.txt"
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]:
```suggestion
    And user "user1" has uploaded file with content "user1 file" to "/randomfile.txt"
```

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 2]
```diff
@@ -648,4 +648,72 @@ Feature: sharing
...
+    When user "user3" shares file "randomfile.txt" with user "user1" with permissions 2 using the sharing API
+    And user "user1" gets the info of the last share using the sharing API
+    Then the fields of the last response should include
+      | uid_owner   | user3              |
+      | share_with  | user1              |
+      | file_target | /randomfile (3).txt|
+      | item_type   | file               |
+      | permissions | 3                  |
+    And as "user1" file "/randomfile (2).txt" should exist
+    And as "user1" file "/randomfile (3).txt" should exist
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]: You could also check the content of each of the 3 files as `user1` - that will make sure that the files really are correctly organized.

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 3]
```diff
@@ -648,4 +648,72 @@ Feature: sharing
...
+    When user "user3" shares folder "zzzfolder" with user "user1" with permissions 16 using the sharing API
+    And user "user1" gets the info of the last share using the sharing API
+    Then the fields of the last response should include
+      | uid_owner   | user3              |
+      | share_with  | user1              |
+      | file_target | /zzzfolder (3)     |
+      | item_type   | folder             |
+      | permissions | 17                 |
+    And as "user1" folder "/zzzfolder (2)" should exist
+    And as "user1" folder "zzzfolder (3)" should exist
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]:
```suggestion
    And as "user1" folder "/zzzfolder (3)" should exist
```

Just to be consistent with the step above (actually with or without `/` works)

File: tests/acceptance/features/apiShareManagementBasic/createShare.feature [hunk 4]
```diff
@@ -648,4 +648,72 @@ Feature: sharing
...
+  Scenario Outline: multiple users share a folder with the same name to a user
+    Given using OCS API version "<ocs_api_version>"
+    And these users have been created with default attributes and without skeleton files:
+      | username |
+      | user1    |
+      | user2    |
+      | user3    |
+    And user "user1" has created folder "/zzzfolder"
+    And user "user2" has created folder "/zzzfolder"
+    And user "user3" has created folder "/zzzfolder"
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareManagementBasic/createShare.feature]: Also create a file in each folder, e.g. with different names.
Then at the end of the scenario check that the corrrect file exists in each folder.

**Comment by phil-davis** [comment]
Also there are conflicts, so pull latest master and rebase and resolve.

**Review by individual-it — APPROVED** [review]
please squash before merging

**Comment by phil-davis** [comment]
PR #35466 was made accidentally against core `master`. So backport is still to be done.

**Comment by HariBhandari07** [comment]
Backport on #35469

**Pull request merged**